assets/javascripts/discourse/components/chat-message-move-to-channel-modal-inner.js

```diff
@@ -0,0 +1,55 @@
+import Component from "@ember/component";
+import discourseComputed from "discourse-common/utils/decorators";
+import { isBlank } from "@ember/utils";
+import { action, computed } from "@ember/object";
+import { ajax } from "discourse/lib/ajax";
+import { inject as service } from "@ember/service";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+
+export default class MoveToChannelModalInner extends Component {
+  @service chat;
+  @service router;
+  tagName = "";
+  sourceChannel = null;
+  destinationChannelId = null;
+  selectedMessageIds = null;
+
+  @computed("destinationChannelId")
+  get disableMoveButton() {
+    return isBlank(this.destinationChannelId);
+  }
+
+  @discourseComputed("chat.publicChannels.[]")
+  availableChannels(publicChannels) {
+    return publicChannels.reject((chan) => chan.id === this.sourceChannel.id);
+  }
+
+  get selectedMessageCount() {
```

Comment:
Hmm does this not need to be a computed property?

Reply:
You can also just use reads() for this

Reply:
I don't think it's necessary to make it computed if the value will not change for the lifetime of the component, I will change to reads() though.